File: SelectionSort.java


public class SelectionSort {

    // Method to perform Selection Sort
    public static void selectionSort(int[] array) {

        // Get array size
        int n = array.length;

        // Move boundary of unsorted array
        for (int i = 0; i < n - 1; i++) {

            // Assume the first unsorted element is the smallest
            int minIndex = i;

            // Find the smallest element in the remaining array
            for (int j = i + 1; j < n; j++) {

                if (array[j] < array[minIndex]) {
                    minIndex = j;
                }

            }

            // Swap the smallest element with the current element
            int temp = array[minIndex];
            array[minIndex] = array[i];
            array[i] = temp;

        }

    }

    // Method to print array
    public static void printArray(int[] array) {

        for (int number : array) {
            System.out.print(number + " ");
        }

        System.out.println();
    }

    public static void main(String[] args) {

        // Unsorted array
        int[] numbers = {29, 10, 14, 37, 13};

        System.out.println("Before Sorting:");
        printArray(numbers);

        // Call Selection Sort
        selectionSort(numbers);

        System.out.println("After Sorting:");
        printArray(numbers);

    }
}

File: SelectionSort.c
#include <stdio.h>

int main() {
    int arr[100];
    int n, i, j;
    int minIndex, temp;

    // Input array size
    printf("Enter the number of elements: ");
    scanf("%d", &n);

    // Input array elements
    printf("Enter %d elements:\n", n);
    for(i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Selection Sort
    for(i = 0; i < n - 1; i++) {

        minIndex = i;

        for(j = i + 1; j < n; j++) {

            if(arr[j] < arr[minIndex]) {
                minIndex = j;
            }

        }

        // Swap the smallest element with the current position
        if(minIndex != i) {
            temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }
    }

    // Display sorted array
    printf("\nSorted Array (Ascending Order):\n");

    for(i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }

    printf("\n");

    return 0;
}
